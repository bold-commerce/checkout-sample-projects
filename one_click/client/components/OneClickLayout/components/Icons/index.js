export { default as TimesCircle } from './TimesCirlce'
export { default as ArrowRight } from './ArrowRight'
export { default as CaretRight } from './CaretRight'