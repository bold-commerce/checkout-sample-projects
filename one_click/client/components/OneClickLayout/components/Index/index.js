export { default as IndexPage } from "./IndexPage";
export { default as IndexGuest } from "./IndexGuest";