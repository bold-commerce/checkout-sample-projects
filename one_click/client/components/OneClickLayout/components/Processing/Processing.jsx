import React from 'react';
import LoadingSpinner from '@boldcommerce/stacks-ui/lib/components/loadingspinner/LoadingSpinner';
import './Processing.scss'

const ProcessingOrder = () => {
    return (
        <div className="processing-page">
            <LoadingSpinner className="processing-spinner" />
            <h1 className="section__title processing-title">
                Processing order...
            </h1>
            <div className="processing-content">
                <p>This may take a few moments. Please remain on the the page until theprocess is complete.</p>
            </div>
        </div>
    )
}

export default ProcessingOrder;