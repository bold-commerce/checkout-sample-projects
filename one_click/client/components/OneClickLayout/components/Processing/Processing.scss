.processing {
    &-page {
        padding: 16px 32px;
        align-items: center;
        justify-content: center;
        display: flex;
        flex-direction: column;
        height: 100%;
    }

    &-spinner {
        width: 32px;
        height: 32px;
    }

    &-title {
        color: rgba(0, 0, 0, 0.4);
        padding: 24px 0px 16px 0px;
        font-size: 24px;
        line-height: 36px;
        font-weight: 700;
    }

    &-content {
        color: rbga(0, 0, 0, 0.65);
        padding: 16px 0px;
        font-size: 14px;
        line-height: 20px;
    }
}