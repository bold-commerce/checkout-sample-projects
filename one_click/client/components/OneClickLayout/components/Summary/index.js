export { default as Summary } from './Summary';
export { default as SummaryCondensed } from './SummaryCondensed';