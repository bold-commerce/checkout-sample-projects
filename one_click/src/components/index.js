export * from './Address';
export * from './BackButton';
export * from './Billing';
export * from './BillingAddress';
export * from './Card';
export * from './CheckoutButton';
export * from './CheckoutForm';
export * from './Confirmation';
export * from './CustomerInfo';
export * from './Discount';
export * from './EmptyState';
export * from './Header';
export * from './Icons';
export * from './Index';
export * from './InventoryItem';
export * from './LineItems';
export * from './LoadingState';
export * from './Payment';
export * from '../pages/Processing';
export * from './Product';
export * from './RadioItem';
export * from './Shipping';
export * from './ShippingLines';
export * from './Summary';
