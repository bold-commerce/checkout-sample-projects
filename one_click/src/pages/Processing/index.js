export { default as ProcessingOrder } from './Processing'