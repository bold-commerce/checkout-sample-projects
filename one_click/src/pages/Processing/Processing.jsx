import React from 'react';
import classNames from 'classnames';
import LoadingSpinner from '@boldcommerce/stacks-ui/lib/components/loadingspinner/LoadingSpinner';
import './Processing.scss'
import { useTranslation } from 'react-i18next';

const ProcessingOrder = ({ show }, ref) => {
    const { t } = useTranslation();
    return (
        <div ref={ref} className={classNames('Sidebar Processing processing-page', show ? 'Sidebar--Show' : 'Sidebar--Hide')}>
            <LoadingSpinner className="processing-spinner" />
            <h1 className="section__title processing-title">
                {t('processing.order')}
            </h1>
            <div className="processing-content">
                <p>{t('processing.description')}</p>
            </div>
        </div>
    )
}

const ProcessingOrderForwardedRef = React.forwardRef(ProcessingOrder);

export default ProcessingOrderForwardedRef;