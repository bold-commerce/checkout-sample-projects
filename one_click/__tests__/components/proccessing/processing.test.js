import React from "react";
import { render } from "@testing-library/react";
import { ProcessingOrder } from '../../../client/components/OneClickLayout/components/Processing';

describe('Processing', () => {
  test('renders Processing component', () => {
    const { asFragment } = render( <ProcessingOrder /> );

    expect(asFragment()).toMatchSnapshot();
  })
})