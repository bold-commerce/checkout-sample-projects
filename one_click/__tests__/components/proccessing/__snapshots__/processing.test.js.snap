// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Processing renders Processing component 1`] = `
<DocumentFragment>
  <div
    class="processing-page"
  >
    <span
      class="stx-loading-spinner processing-spinner"
    />
    <h1
      class="section__title processing-title"
    >
      Processing order...
    </h1>
    <div
      class="processing-content"
    >
      <p>
        This may take a few moments. Please remain on the the page until the process is complete.
      </p>
    </div>
  </div>
</DocumentFragment>
`;
